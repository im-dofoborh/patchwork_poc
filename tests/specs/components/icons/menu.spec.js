import { mount, createLocalVue } from '@vue/test-utils';
import MenuIcon from '~/components/icons/Menu';

describe('MenuIcon component', () => {
  let localVue = null;
  let wrapper = null;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.component('menu-icon', MenuIcon);
    wrapper = mount(MenuIcon, {
      localVue,
    });
  })

  it('renders default variant', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders open variant', () => {
    wrapper.setProps({
      open: true,
    })
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('emits click event on click', () => {
    wrapper.trigger('click');
    expect(wrapper.emitted().click).toBeTruthy();
  });
});
