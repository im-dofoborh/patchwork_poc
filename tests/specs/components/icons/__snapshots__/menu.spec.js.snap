// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuIcon component renders default variant 1`] = `
<div class="sg-menu-icon">
  <div class="sg-menu-icon__bar"></div>
  <div class="sg-menu-icon__bar"></div>
  <div class="sg-menu-icon__bar"></div>
</div>
`;

exports[`MenuIcon component renders open variant 1`] = `
<div class="sg-menu-icon sg-menu-icon--open">
  <div class="sg-menu-icon__bar"></div>
  <div class="sg-menu-icon__bar"></div>
  <div class="sg-menu-icon__bar"></div>
</div>
`;
